core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/BaseDefaultHeartbeatPropertyProvider.java

```diff
@@ -0,0 +1,162 @@
+package com.microsoft.applicationinsights.internal.heartbeat;
+
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.internal.util.PropertyHelper;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+/**
+ * <h1>Base Heartbeat Property Provider</h1>
+ *
+ * <p>
+ *   This class is a concrete implementation of {@link com.microsoft.applicationinsights.internal.heartbeat.HeartBeatDefaultPayloadProviderInterface}
+ *   It enables setting SDK Metadata to heartbeat payload.
+ * </p>
+ *
+ * @author Dhaval Doshi
+ * @since 03-30-2018
+ */
+public class BaseDefaultHeartbeatPropertyProvider implements HeartBeatDefaultPayloadProviderInterface {
+
+  /**
+   * Collection holding default properties for this default provider.
+   */
+  private final Set<String> defaultFields;
+
+  /**
+   * Random GUID that would help in analysis when app has stopped and restarted. Each restart will
+   * have a new GUID. If the application is unstable and goes through frequent restarts this will help
+   * us identify instability in the analytics backend.
+   */
+  private static UUID uniqueProcessId;
+
+  /**
+   * Name of this provider.
+   */
+  private final String name = "Base";
+
+  public BaseDefaultHeartbeatPropertyProvider() {
+    defaultFields = new HashSet<>();
+    initializeDefaultFields(defaultFields);
+  }
+
+  @Override
+  public String getName() {
+    return this.name;
+  }
+
+  @Override
+  public boolean isKeyWord(String keyword) {
+    return defaultFields.contains(keyword);
+  }
+
+  @Override
+  public Callable<Boolean> setDefaultPayload(final List<String> disableFields,
+      final HeartBeatProviderInterface provider) {
+    return new Callable<Boolean>() {
+
+      // using volatile here to avoid caching in threads.
+      volatile boolean hasSetValues = false;
```

Comment:
This doesn't need to be volatile. You're reasoning in the comment is incorrect. No other thread can change this value. In fact, these should just be local to the `call()` method to make it a bit easier for GC to clean it up.